### PR TITLE
some refactor of intent fireing

### PR DESCRIPTION
**Pull request by MaximeZmt**

### CODE REVIEW

File: app/src/main/java/ch/sdp/vibester/helper/IntentSwitcher.kt [hunk 1]
```diff
@@ -2,12 +2,18 @@ package ch.sdp.vibester.helper
...
 class IntentSwitcher private constructor() {
     companion object {
-        fun switchBackToWelcome(ctx: Context) {
-            ctx.startActivity(Intent(ctx, WelcomeActivity::class.java))
+        fun switch(ctx: Context, arg: Class<*>?, extrasMap: Map<String, Serializable>?) {
+            val intent = Intent(ctx, arg)
+            if (extrasMap != null) {
+                extrasMap.forEach{
+                        entry -> intent.putExtra(entry.key, entry.value)
+                }
```
Comment (kamilababayeva) [inline app/src/main/java/ch/sdp/vibester/helper/IntentSwitcher.kt:14]: Can we test this?

File: app/src/main/java/ch/sdp/vibester/helper/GameManager.kt [hunk 2]
```diff
@@ -78,7 +81,11 @@ open class GameManager : Serializable {
      */
     @JvmName("getCurrentSong1")
     fun getCurrentSong(): Song {
-        return currentSong
+        if(this::currentSong.isInitialized) {
+            return currentSong
+        } else {
+            return Song.songBuilder("http://example.com", "http://example.com", songName, artistName)
```
Comment (kamilababayeva) [inline app/src/main/java/ch/sdp/vibester/helper/GameManager.kt:87]: what is the purpose if this song.builder?

File: app/src/main/java/ch/sdp/vibester/helper/IntentSwitcher.kt [hunk 3]
```diff
@@ -2,12 +2,18 @@ package ch.sdp.vibester.helper
...
 import android.content.Context
 import android.content.Intent
-import ch.sdp.vibester.activity.WelcomeActivity
+import java.io.Serializable
 
 class IntentSwitcher private constructor() {
     companion object {
-        fun switchBackToWelcome(ctx: Context) {
-            ctx.startActivity(Intent(ctx, WelcomeActivity::class.java))
+        fun switch(ctx: Context, arg: Class<*>?, extrasMap: Map<String, Serializable>?) {
```
Comment (kamilababayeva) [inline app/src/main/java/ch/sdp/vibester/helper/IntentSwitcher.kt:9]: we can put extrasmap as null by default, so we dont seutp null in each call of welcome activity

File: app/src/main/java/ch/sdp/vibester/activity/LyricsBelongGameActivity.kt [hunk 4]
```diff
@@ -224,20 +224,12 @@ class LyricsBelongGameActivity : GameActivity() {
...
-    fun getSongName(): String {
-        return songName
+    fun testGetAndCheckLyrics(ctx: Context, song: Song, speechInput: String, gameManager: GameManager) {
+        getAndCheckLyrics(ctx, song, speechInput, gameManager)
     }
 
-    fun getArtistName(): String {
-        return artistName
+    fun getSong(): Song {
+        return song
```
Comment (kamilababayeva) [inline app/src/main/java/ch/sdp/vibester/activity/LyricsBelongGameActivity.kt:232]: can we test it?

**Review by kamilababayeva — COMMENTED** [review]
I have some questions regarding testing some lines!

Thank you for your changes! Looks good to me

File: app/src/main/java/ch/sdp/vibester/helper/GameManager.kt (hunk 2 above)
Comment (MaximeZmt) [reply app/src/main/java/ch/sdp/vibester/helper/GameManager.kt:87]: The purpose of this song builder is that when the currentSong is not initialized yet (mean no web connection for instance/low connection), it does not crash and has song title and name

File: app/src/main/java/ch/sdp/vibester/helper/IntentSwitcher.kt (hunk 1 above)
Comment (MaximeZmt) [reply app/src/main/java/ch/sdp/vibester/helper/IntentSwitcher.kt:14]: of course

File: app/src/main/java/ch/sdp/vibester/helper/IntentSwitcher.kt (hunk 3 above)
Comment (MaximeZmt) [reply app/src/main/java/ch/sdp/vibester/helper/IntentSwitcher.kt:9]: yes

File: app/src/main/java/ch/sdp/vibester/activity/LyricsBelongGameActivity.kt (hunk 4 above)
Comment (MaximeZmt) [reply app/src/main/java/ch/sdp/vibester/activity/LyricsBelongGameActivity.kt:232]: yes

File: app/src/main/java/ch/sdp/vibester/activity/AuthenticationActivity.kt [hunk 5]
```diff
@@ -99,7 +99,7 @@ class AuthenticationActivity : AppCompatActivity() {
      * Listener bound to the red return button in the Authentication activity.
      */
     fun returnToMainListener(view: View) {
-        IntentSwitcher.switchBackToWelcome(this)
+        IntentSwitcher.switch(this, WelcomeActivity::class.java, null)
```
Comment (laurislopata) [inline app/src/main/java/ch/sdp/vibester/activity/AuthenticationActivity.kt:102]: Maybe a small comment to expain why we pass null here?

**Review by laurislopata — APPROVED** [review]
LGTM, just a tiny comment but I think you can add it later if you want :)

File: app/src/main/java/ch/sdp/vibester/activity/LyricsBelongGameActivity.kt (hunk 4 above)
Comment (MaximeZmt) [reply app/src/main/java/ch/sdp/vibester/activity/LyricsBelongGameActivity.kt:232]: in fact it is not used so removed

File: app/src/main/java/ch/sdp/vibester/activity/AuthenticationActivity.kt (hunk 5 above)
Comment (MaximeZmt) [reply app/src/main/java/ch/sdp/vibester/activity/AuthenticationActivity.kt:102]: I've removed the null, not it is by default

File: app/src/main/java/ch/sdp/vibester/helper/GameManager.kt (hunk 2 above)
Comment (kamilababayeva) [reply app/src/main/java/ch/sdp/vibester/helper/GameManager.kt:87]: I am not sure whether we should put a fake song. maybe we can put some text to screen. lets fix it next pr!

**Review by kamilababayeva — APPROVED** [review]

**Pull request merged**